pkg/controller/garbagecollector/garbagecollector.go

```diff
@@ -124,29 +164,33 @@ func (p *Propagator) processEvent() {
 		return
 	}
 	defer p.eventQueue.Done(key)
-	event, ok := key.(Event)
+	event, ok := key.(event)
 	if !ok {
-		glog.Errorf("expect an Event, got %v", key)
+		glog.Errorf("expect an event, got %v", key)
 		return
 	}
 	obj := event.Obj
 	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		glog.Errorf("cannot access obj: %v", err)
+		return
+	}
 	typeAccessor, err := meta.TypeAccessor(obj)
 	if err != nil {
 		glog.Errorf("cannot access obj: %v", err)
 		return
 	}
 	glog.V(6).Infof("Propagator process object: %s/%s, namespace %s, name %s, event type %s", typeAccessor.GetAPIVersion(), typeAccessor.GetKind(), accessor.GetNamespace(), accessor.GetName(), event.Type)
 	switch event.Type {
-	case Add:
+	case addEvent:
 		// Check if the node already exsits
-		_, ok := p.graphLookup[accessor.GetUID()]
+		_, ok := p.uidToNode[accessor.GetUID()]
 		if ok {
 			// We can optionally check if the node in the graph is in sync with that in the Store.
```

Comment:
hm..the Update handler will need to implement the orphan finalizers in the future, which is not required by the Add handler. I prefer keep them separate.